tools/clang/test/CodeGenSPIRV/texture.array.gather-alpha.hlsl

```diff
@@ -44,5 +49,42 @@ float4 main(float3 location: A) : SV_Target {
 // CHECK-NEXT:                       OpStore %d [[res]]
     uint4 d = t2u4.GatherAlpha(gSampler, location, int2(1, 2), int2(3, 4), int2(5, 6), int2(7, 8));
 
+    uint status;
+// CHECK:              [[t2u4:%\d+]] = OpLoad %type_2d_image_array_0 %t2u4
+// CHECK-NEXT:     [[gSampler:%\d+]] = OpLoad %type_sampler %gSampler
+// CHECK-NEXT:          [[loc:%\d+]] = OpLoad %v3float %location
+// CHECK-NEXT:   [[sampledImg:%\d+]] = OpSampledImage %type_sampled_image_0 [[t2u4]] [[gSampler]]
+// CHECK-NEXT: [[structResult:%\d+]] = OpImageSparseGather %SparseResidencyStruct [[sampledImg]] [[loc]] %int_3 ConstOffset [[c12]]
+// CHECK-NEXT:       [[status:%\d+]] = OpCompositeExtract %uint [[structResult]] 0
+// CHECK-NEXT:                         OpStore %status [[status]]
+// CHECK-NEXT:       [[result:%\d+]] = OpCompositeExtract %v4uint [[structResult]] 1
+// CHECK-NEXT:                         OpStore %e [[result]]
+    uint4 e = t2u4.GatherAlpha(gSampler, location, int2(1, 2), status);
+
+// CHECK:              [[t2u4:%\d+]] = OpLoad %type_2d_image_array_0 %t2u4
+// CHECK-NEXT:     [[gSampler:%\d+]] = OpLoad %type_sampler %gSampler
+// CHECK-NEXT:          [[loc:%\d+]] = OpLoad %v3float %location
+// CHECK-NEXT:   [[sampledImg:%\d+]] = OpSampledImage %type_sampled_image_0 [[t2u4]] [[gSampler]]
+// CHECK-NEXT: [[structResult:%\d+]] = OpImageSparseGather %SparseResidencyStruct [[sampledImg]] [[loc]] %int_3 ConstOffsets [[c1to8]]
+// CHECK-NEXT:       [[status:%\d+]] = OpCompositeExtract %uint [[structResult]] 0
+// CHECK-NEXT:                         OpStore %status [[status]]
+// CHECK-NEXT:       [[result:%\d+]] = OpCompositeExtract %v4uint [[structResult]] 1
+// CHECK-NEXT:                         OpStore %f [[result]]
+    uint4 f = t2u4.GatherAlpha(gSampler, location, int2(1, 2), int2(3, 4), int2(5, 6), int2(7, 8), status);
+
+// NOTE: The HLSL reference states that the location parameter should be a scalar float.
```

Comment:
I think the doc is wrong. It should take 3-component to index into Cube, and 4-component to index into CubeArray. We can remove the note here and follow the DXC implementation.

Similar for the rest.

Reply:
I see... Sounds good. Thanks. I will make these changes.